vars/buildFailureMessage.groovy

```diff
@@ -0,0 +1,17 @@
+@NonCPS
+def call(){
+    def ERROR_STRING = "ERROR"
+    def message = ""
+    String delimiter = ","
+    def performance_log = currentBuild.rawBuild.getLog(Integer.MAX_VALUE).join('\n')
+    performance_log.eachLine() { line ->
+        if (line.matches(".*$ERROR_STRING.*")) {
+            message+=(line+delimiter)
```

Comment:
This message can be long if there are a log of errors. Let's make it a `StringBuffer` or a `StringBuilder` instead

Reply:
Thats a good point, Added Reader reading character streams and pushed in my latest commit.